src/main/java/teammates/common/util/Config.java

```diff
@@ -95,7 +95,11 @@ private String getGcsBucketname() {
     }
 
     private String getAppUrl() {
-        return props.getProperty("app.url");
+        return Url.trimTrailingSlash(props.getProperty("app.url"));
+    }
+    
+    public static AppUrl getAppUrl(String url) {
+        return new AppUrl(APP_URL, url);
     }
```

Comment:
I don't really get this requirement. If it returns an `AppUrl` object with no relative URL, then what's the use of it? And how would this be a factory method, then?